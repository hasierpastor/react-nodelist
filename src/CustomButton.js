import React from 'react';
import { Button } from 'reactstrap';

const buttonStyle = {
  margin: '1.5rem',
  padding: '1rem'
};

//Created a very simple button function component so that it would be reusable in different apps/situations
//When button is clicked it will call props.handleClick which is bound to parent and will change state in parent
//In this case we will use handleClick to sort nodes and re-render node table
function CustomButton(props) {
  return (
    <Button
      outline
      style={buttonStyle}
      color="success"
      onClick={props.handleClick}
    >
      {props.text}
    </Button>
  );
}

export default CustomButton;
