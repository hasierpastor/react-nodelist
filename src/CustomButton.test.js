import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomButton from './CustomButton';

it('renders without crashing', function() {
  shallow(<CustomButton />);
});

it('matches snapshot', function() {
  let wrapper = shallow(<CustomButton />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
