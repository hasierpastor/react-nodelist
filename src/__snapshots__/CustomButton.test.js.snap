// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Button
  color="success"
  outline={true}
  style={
    Object {
      "margin": "1.5rem",
      "padding": "1rem",
    }
  }
  tag="button"
/>
`;
